app/options.go

```diff
@@ -127,9 +125,13 @@ func StartSearchEngine(s *Server) error {
 	return nil
 }
 
+// SetLogger requires platform service to be initialized before calling.
+// If not, logger should be set after platform service are initialized.
 func SetLogger(logger *mlog.Logger) Option {
 	return func(s *Server) error {
-		s.Log = logger
+		if s.platform != nil {
+			s.platform.SetLogger(logger)
+		}
```

Comment:
Maybe it would be better return error here as we were not be able to set logger anyway. Thoughts?

Reply:
This is just an internal function used in tests right? Feel free to take a call.